students/BrandonHenson/Lesson9/Mailroom_5.py

```diff
@@ -0,0 +1,138 @@
+# Brandon Henson
+# 5/17/18
+# Lesson 09 mailroom_5.py
+# Refactoring
+
+
+class Donor(object):
+
+    donor_history = {'Brandon Henson': [1005.49, 3116.72, 5200],
+                        'Alicia Henson': [21.47, 1500],
+                        'Michael Green': [2400.54],
+                        'Brandon Henson Jr': [355.42, 579.31],
+                        'Kaiya Henson': [636.9, 850.13, 125.23]}
+
+    name_list = [i for i in donor_history]
+
+    def __init__(self, name, *donation):
+        self.donations = []
+        self.name = name
+        self.donation = donation
+        for i in self.donation:
+            self.donations.append(i)
+        Donor.name_list.append(self.name)
+        Donor.donor_history[self.name] = self.donation
+
+    @property
+    def total(self):
+        return sum(self.donations)
+
+    @property
+    def average(self):
+        return self.total/self.num_of_donations
+
+    @property
+    def num_of_donations(self):
+        return len(self.donations)
+
+
+class All(Donor):
+    def __init__(self):
+        pass
+
+    def all_total(self):
+        list_of_sums = []
+        count = 0
+        for k, v in Donor.donor_history.items():
+            for i in v:
+                count = count+1
+            total = sum(v)
+            list_of_sums.append(total)
+        gt = (sum(list_of_sums))
+        av = gt/count
+        gt = str(round(gt, 2))
+        av = str(round(av, 2))
+        print('The grand total of all donations is ${}.\
+ The average amount per donation is ${}.'.format(gt, av))
```

Comment:
This method is static (doesn't reference instantiated `self` object) which is also a good indication that you don't need this class at all, since it's the only method on the class. I don't see any reason to subclass `Donor` here, this would be more appropriately defined as an additional method on the `Donor` class